pkgs/applications/science/logic/protege/default.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv
+, fetchzip
+, jre
+, makeWrapper
+, makeDesktopItem
+, iconConvTools
+}:
+
+stdenv.mkDerivation rec {
+  pname = "protege";
+  version = "5.5.0";
+
+  src = fetchzip {
+    url = "https://github.com/protegeproject/protege-distribution/releases/download/v${version}/Protege-${version}-platform-independent.zip";
+    sha256 = "1v82ph1pqvnc1qynhiapzw0jwm9rphsb580lc96zwsvhrr0wd690";
+  };
+
+  nativeBuildInputs = [ makeWrapper iconConvTools ];
+
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname;
+    desktopName = "Protege";
+    genericName = meta.description;
+    comment = meta.description;
+    categories = "Development;";
+    icon = pname;
+    extraEntries = ''
+      StartupWMClass=${pname}
+    '';
+  };
+
+  # makeWrapper flags adapted from protege's run.sh script (see extracted sources)
+  installPhase = ''
+    mkdir -p $out/src
```

Comment:
Please add proper `runHook preInstall` and `runHook postInstall`. Per https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md .